packages/block-editor/src/components/block-title/index.js

```diff
@@ -44,22 +49,16 @@ export default function BlockTitle( { clientId } ) {
 		[ clientId ]
 	);
 
-	if ( ! name ) {
-		return null;
-	}
-
+	const blockInformation = useMatchingVariationInformation( clientId );
+	if ( ! blockInformation ) return null;
 	const blockType = getBlockType( name );
-	if ( ! blockType ) {
-		return null;
-	}
-
-	const { title } = blockType;
 	const label = getBlockLabel( blockType, attributes );
-
-	// Label will often fall back to the title if no label is defined for the
+	// Label will fallback to the title if no label is defined for the
 	// current label context. We do not want "Paragraph: Paragraph".
-	if ( label !== title ) {
-		return `${ title }: ${ truncate( label, { length: 15 } ) }`;
+	// If label is defined we prioritize it over possible possible
+	// block variation match title.
+	if ( label !== blockType.title ) {
+		return `${ blockType.title }: ${ truncate( label, { length: 15 } ) }`;
```

Comment:
Should it use the block information here as well?

```suggestion
		return `${ blockInformation.title }: ${ truncate( label, { length: 15 } ) }`;
```

Reply:
I'm not really sure.. I want to revisit all this `label` functionality, but I want to do it in a follow up to investigate properly and not include possibly even more changes in this PR.

For now by leaving it as is `blockType.title` it keeps the exact previous functionality for cases that `label` was used.

Reply:
Sure, I just wanted to flag it.